packages/fuels-contract/src/contract.rs

```diff
@@ -512,16 +527,40 @@ where
         .await
     }
 
+    /// Simulates calls and attempts to set output variables based on the received error.
+    /// Forwards the reiceived error if it cannot be fixed.
+    pub async fn auto_setup(mut self, max_attempts: Option<u64>) -> Self {
```

Comment:
`auto_setup` isn't a good name. It doesn't tell me _what_ it is automatically being set up; I have to read the doc-level comment to know that. It gets extra confusing once the user has to reason about the "automatically set up of something" and the need for a maximum number of attempts, e.g., "why does it need to do many attempts just to set up?!".

It's better to give it a name that makes sense to have `max_attemps`. Probably with `"estimate"` in it. I understand we don't want to put `"output variables"` in it because this will likely be used for estimating the contract input set as well. So... I propose `set_estimated_tx_dependencies`:
1. `set` hints that it's mutating something;
2. `estimated` because it's an estimation (and it takes `max_attemps` attempts);
3. `tx_dependencies` because it's doing this operation on the dependencies needed for this transaction to work (output variables and, soon, the contract input set);
4. We're `set`ting the `estimated` `tx_dependencies` for the current transaction.

Open to suggestions/improvements on it, of course.

Reply:
It's a better name.

It's just the `set` part that rubs me wrong. We are setting it, sure, but we're also doing the estimating. 

Perhaps it's just that I'm not used to seeing a "setter" do such work. 

But that means we're to split the estimation and setting -- which causes issues of its own. 

Or use action-y names like `estimate_tx_deps` but that doesn't have the set in it. You'd have to see it takes a `&mut self`

Reply:
Yeah, I'm not a huge fan of `set`, but it does tell us that it's mutating; that's its sole purpose here.

But you raise a good point: `estimate_tx_deps` is better because no `set`, but if a dev see it takes a `&mut self`... then it's pretty obvious it's mutating. At least one would hope it's pretty obvious haha.

So I'd say let's go with `estimate_tx_deps`. Also, are we sure there's no ambiguity/confusion between `deps` and `dependencies`?

Reply:
I was thinking the same, but I figured I prefer it to state that it is setting something.